test/t8_forest/t8_gtest_element_is_boundary.cxx

```diff
@@ -0,0 +1,216 @@
+    if (t8_cmesh_is_empty (cmesh)) {
+      /* forest_commit does not support empty cmeshes, we skip this case */
+      GTEST_SKIP ();
+    }
+    /* Build the default scheme (TODO: Test this with all schemes) */
+    scheme = t8_scheme_new_default_cxx ();
+    forest = t8_forest_new_uniform (cmesh, scheme, level, 0, sc_MPI_COMM_WORLD);
+    t8_forest_ref (forest);
+    //const int maxlevel = t8_forest_get_maxlevel (forest);
+    int maxlevel = 7;
+    const int recursive_adapt = 1;
+    forest_adapt = t8_forest_new_adapt (forest, t8_test_adapt_first_child, recursive_adapt, 0, &maxlevel);
+  }
+
+  void
+  TearDown () override
+  {
+    GTEST_SKIP ();
+    if (t8_cmesh_is_empty (cmesh)) {
+      t8_cmesh_destroy (&cmesh);
+    }
+    else {
+      t8_forest_unref (&forest);
+      t8_forest_unref (&forest_adapt);
+    }
+  }
+
+  t8_cmesh_t cmesh;
+  t8_forest_t forest, forest_adapt;
+  t8_scheme_cxx_t *scheme;
+};
+
+void
+t8_test_element_is_boundary_for_forest (t8_forest_t forest, t8_cmesh_t cmesh,
+                                        const int each_element_face_is_expected_boundary)
+{
+  const t8_locidx_t num_local_trees = t8_forest_get_num_local_trees (forest);
+
+  for (t8_locidx_t itree = 0; itree < num_local_trees; ++itree) {
+    const t8_locidx_t num_elements_in_tree = t8_forest_get_tree_num_elements (forest, itree);
+    const t8_eclass_t tree_class = t8_forest_get_tree_class (forest, itree);
+    const t8_eclass_scheme_c *scheme = t8_forest_get_eclass_scheme (forest, tree_class);
+    /* Iterate over all the tree's leaf elements, check whether the leaf
+     * is correctly identified by t8_forest_element_is_boundary,
+     * build its parent and its first child (if they exist), and verify
+     * that t8_forest_element_is_boundary returns false. */
+    for (t8_locidx_t ielement = 0; ielement < num_elements_in_tree; ++ielement) {
+      const t8_element_t *leaf_element = t8_forest_get_element_in_tree (forest, itree, ielement);
+      const int num_element_faces = scheme->t8_element_num_faces (leaf_element);
+      for (int iface = 0; iface < num_element_faces; ++iface) {
+        /* Iterate over all faces */
+        int face_is_at_boundary = 0;
+        if (!each_element_face_is_expected_boundary) {
+          /* Manually check whether this element is at the boundary */
+          if (scheme->t8_element_is_root_boundary (leaf_element, iface)) {
+            /* This face is at a tree boundary, so it might be at the domain boundary. */
+            /* We check whether the tree is at the domain boundary. */
+            const int tree_face = scheme->t8_element_tree_face (leaf_element, iface);
+            const t8_locidx_t cmesh_local_tree = t8_forest_ltreeid_to_cmesh_ltreeid (forest, itree);
+            if (t8_cmesh_tree_face_is_boundary (cmesh, cmesh_local_tree, tree_face)) {
+              /* This element is at the domain boundary. */
+              face_is_at_boundary = 1;
+            }
+          }
```

Comment:
This test is mainly copying implementation from the actual functionality:
```
/* Check whether this leaf is at the boundary of its tree. */
  const int is_root_boundary = scheme->t8_element_is_root_boundary (leaf, face);

  if (is_root_boundary) {
    /* This leaf is at a tree's boundary. It is*/
    const int cmesh_face = scheme->t8_element_tree_face (leaf, face);
    const t8_cmesh_t cmesh = t8_forest_get_cmesh (forest);
    const t8_locidx_t cmesh_local_tree = t8_forest_ltreeid_to_cmesh_ltreeid (forest, local_tree);
    int tree_boundary = t8_cmesh_tree_face_is_boundary (cmesh, cmesh_local_tree, cmesh_face);
    if (tree_boundary) {
      return 1;
    }
  }
```
I am not sure about its effectiveness.

Reply:
I do not advocate for removing it, because it ensures that a different implementation cannot break other stuff, but ideally we also have test, where we build a small forest, where we know how the function should behave

Reply:
i added a new test with a single element forest for each eclass.
Every face of that element has to be at the boundary.